rnd/autogpt_server/autogpt_server/executor/manager.py

```diff
@@ -336,6 +338,14 @@ def on_node_executor_start(cls):
         cls.loop.run_until_complete(db.connect())
         cls.agent_server_client = get_agent_server_client()
 
+        # Set up shutdown handler
+        signal.signal(signal.SIGTERM, lambda _, __: cls.node_executor_stop())
+
+    @classmethod
+    def node_executor_stop(cls):
```

Comment:
nit: can we keep the same naming convention (event), e.g: `on_node_executor_stop` ?

Reply:
Is the exit that triggers the sigterm?

Reply:
* Changed to cleanup triggered by `atexit`
* Renamed cleanup function to `on_(node|graph)_executor_stop`

Reply:
I had to add back `stop_node_executor` to handle SIGTERM. A SIGTERM handler must call `sys.exit(0)`, while an `atexit` handler shouldn't.